app/views/controllers/application/app/_banners.html.erb

```diff
@@ -9,28 +9,26 @@
     <div id="admin_banner" class="h3 text-center font-weight-bold p-2">
       DANGER: You are currently logged in as <%= User.current.login %>.
     </div>
-  <% end
+  <% end %>
 
-  banner_text = :app_banner_box.t
-  banner_time = TranslationString.banner_time.utc.strftime("%Y%m%d%H%M%S")
-  cookie_time = cookies["hideBanner2"].to_s
-  dismissed   = cookie_time.match(/^\d{14}$/) &&
-                cookie_time.to_d > banner_time.to_d - 10000
-  if banner_text.to_s.length > 1 && !dismissed %>
-    <div id="message_banner" class="alert alert-success message-banner"
-         data-controller="banner">
+  <div data-controller="banner">
+    <div id="banner" class="alert alert-success message-banner" data-banner-target="banner">
       <button
-        type="button"
-        class="close"
-        data-dismiss="alert"
-        data-action="banner#setCookie"
-        data-banner-time-param="<%= banner_time %>"
-        aria-label="Close">
-        <span aria-hidden="true">&times;</span>
+	class="close"
+	id="dismiss-banner"
+	data-banner-target="dismissButton"
+	aria-label="Close"
+	data-version="<%= Banner.current.version %>">
+        <span class="glyphicon glyphicon-chevron-up" aria-hidden="true"></span>
       </button>
-      <%= banner_text %>
+      <p><%= Banner.current.message.t %></p>
     </div>
-  <% end %>
 
+    <div id="show-banner-container">
```

Comment:
See stimulus recommendation below.
```suggestion
    <div id="show-banner-container" data-banner-target="container">
```

Reply:
Interesting.  Why do you use underscores for class names?